bin/upgrade

```diff
@@ -126,17 +163,32 @@ function handle_git_update() {
   echo "Checking for code update..."
 
   if ! git_pull_available "$current_branch"; then
-    echo "No code update available"
+    echo "No code update available for download"
   else
-    echo "Code update available! (current commit is $current_commit)"
+    echo "Code update available for download!"
+
+    local should_fetch="n"
+    read -r -p "Proceed to download update? [y/n] " should_fetch
```

Comment:
I think these two prompts could be confusing for users, as they might not appreciate the difference between fetch and pull in git-land. Plus, prompting for the initial fetch feels a bit redundant.

I suggest we:
- automatically fetch from origin
- display the changelog
- _prompt_ the user, ask if they want to merge the changes

If (at any stage) there are no changes to pull, we can just stop with an appropriate message.

Otherwise, I think this is a great improvement :tada:

Reply:
> I suggest we:
> 
> * automatically fetch from origin
> * display the changelog
> * _prompt_ the user, ask if they want to merge the changes

Done in https://github.com/overleaf/toolkit/pull/27/commits/4b48169a56e6531d1cd55334150b874f16f7119d

> If (at any stage) there are no changes to pull, we can just stop with an appropriate message.

The user could dismiss the apply step (downloaded changes on remote branch). We should check for local changes either way (as is currently).

Reply:
... and thanks for the feedback :)